RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h

```diff
@@ -0,0 +1,36 @@
+#ifndef __RecoLocalCalo_HGCalRecAlgos_RecHitTools_h__
+#define __RecoLocalCalo_HGCalRecAlgos_RecHitTools_h__
+
+#include <array>
+#include <cmath>
+
+class HGCalGeometry;
+class HGCalDDDConstants;
+class DetId;
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}
+
+namespace hgcal {
+  class RecHitTools {    
```

Comment:
One thing to keep in mind, ESProducts can not depend on the Event and all their methods must be const since multiple threads will talk to them concurrently.